Classes/KIFUIViewTestActor.m

```diff
@@ -0,0 +1,396 @@
+//
+//  KIFUIViewActor.m
+//  KIF
+//
+//  Created by Alex Odawa on 1/21/15.
+//
+//
+
+#import "KIFUIViewTestActor.h"
+#import "UIWindow-KIFAdditions.h"
+
+
+@interface KIFUIViewTestActor ()
+
+@property (nonatomic, strong, readonly) KIFUITestActor *actor;
+@property (nonatomic, strong, readwrite) NSPredicate *predicate;
+
+@end
+
+
+@implementation KIFUIViewTestActor
+
+#pragma mark - Initialization
+
+- (instancetype)usingPredicate:(NSPredicate *)predicate;
+{
+
+    [self _appendPredicate:predicate];
+    return  self;
+}
+
+- (instancetype)usingAccessibilityLabel:(NSString *)accessibilityLabel;
+{
+    int systemVersion = [UIDevice currentDevice].systemVersion.intValue;
+
+    if ([accessibilityLabel rangeOfString:@"\n"].location == NSNotFound || systemVersion == 6) {
+        return [self usingPredicate:[NSPredicate predicateWithFormat:@"accessibilityLabel MATCHES %@", accessibilityLabel]];
+    }
+
+    // On iOS 6 the accessibility label may contain line breaks, so when trying to find the
+    // element, these line breaks are necessary. But on iOS 7 the system replaces them with
+    // spaces. So the same test breaks on either iOS 6 or iOS 7. iOS 8 befuddles this again by
+    // limiting replacement to spaces in between strings.
+    // UNLESS the accessibility label is set programatically in which case the line breaks remain regardless of OS version.
+    // To work around this replace the line breaks using the preferred method and try matching both.
+
+    //this feels horribly hacky and looks bad in our - (NSString *)description :( but tests are passing
+    // We might consider replaceing the predicate with a block that can do cleaner checking,
+
+    NSString *alternate = nil;
+    if (systemVersion == 7) {
+        alternate = [accessibilityLabel stringByReplacingOccurrencesOfString:@"\n" withString:@" "];
+    } else {
+        alternate = [accessibilityLabel stringByReplacingOccurrencesOfString:@"\\b\\n\\b" withString:@" " options:NSRegularExpressionSearch range:NSMakeRange(0, accessibilityLabel.length)];
+    }
+
+    return [self usingPredicate:[NSPredicate predicateWithFormat:@"accessibilityLabel MATCHES %@ OR accessibilityLabel MATCHES %@", accessibilityLabel, alternate]];
+}
+
+- (instancetype)usingAccessibilityIdentifier:(NSString *)accessibilityIdentifier;
+{
+    return [self usingPredicate:[NSPredicate predicateWithFormat:@"accessibilityIdentifier MATCHES %@", accessibilityIdentifier]];
+}
+
+- (instancetype)usingTraits:(UIAccessibilityTraits)traits;
+{
+    return [self usingPredicate:[NSPredicate predicateWithFormat:@"(accessibilityTraits & %@) == %@", @(traits), @(traits)]];
+}
+
+- (instancetype)usingValue:(NSString *)value;
+{
+    return [self usingPredicate:[NSPredicate predicateWithFormat:@"accessibilityValue MATCHES %@", value]];
+}
+
+- (instancetype)usingExpectedClass:(Class)expectedClass;
+{
+    return [self usingPredicate:[NSPredicate predicateWithFormat:@"class MATCHES %@", expectedClass]];
```

Comment:
Block predicates create an issue with our error logging as their `description` isn't very usefull.  You're absolutely right though, this is incorrect and wont work with subclasses, etc.

Thinking about it more, I dont think we should support this at all. I had initially implemented it to debug an issue where KIF was finding the wrong element, but the whole concept of a class is an implementation detail and  isn't user-facing, so it really doesnt belong. If needed anyone can build a class predicate and pass it into `usingPredicate:`